pkg/utils/segutils.go

```diff
@@ -139,3 +141,17 @@ func IsSubWordPresent(haystack []byte, needle []byte) bool {
 
 	return false
 }
+
+func IsPatternPresent(haystack []byte, pattern []byte) bool {
+	// Convert pattern to string and replace '*' with '.*' to create a regex pattern
+	regexPattern := strings.Replace(string(pattern), "*", ".*", -1)
+
+	// Compile the regex
+	r, err := regexp.Compile(regexPattern)
```

Comment:
Doing regex compile on every check is very expensive. This should be done once . Take a look at the other applycsg function and how it passes the qvaldte which has the compiled regex in it

Reply:
Got it! I'll use the compiled regex from qvaldte